src/material/button/_button-base.scss

```diff
@@ -133,3 +133,48 @@
     }
   }
 }
+
+@mixin mat-private-button-horizontal-layout($prefix, $slots, $has-with-icon-padding) {
+  @include token-utils.use-tokens($prefix, $slots) {
+    $icon-spacing: token-utils.get-token-variable-reference(icon-spacing, true);
+    $icon-offset: token-utils.get-token-variable-reference(icon-offset, true);
+    $horizontal-padding: token-utils.get-token-variable-reference(horizontal-padding, true);
+    padding: 0 $horizontal-padding;
```

Comment:
I think we can use `padding-inline` now, which would allow you to do this with just `create-token-slot()`

Reply:
I wrote it this way to minimize the difference with MDC. `padding-inline` will probably break existing overrides.